src/bsoncxx/document/element.hpp

```diff
@@ -390,6 +352,25 @@ class BSONCXX_API element {
     array::element operator[](std::uint32_t i) const;
 
    private:
+    ///
+    /// Construct an element as an offset into a buffer of bson bytes.
+    ///
+    /// @param raw
+    ///   A pointer to the raw bson bytes.
+    ///
+    /// @param length
+    ///   The size of the bson buffer.
+    ///
+    /// @param offset
+    ///   The element's offset into the buffer.
+    ///
+    BSONCXX_PRIVATE explicit element(const std::uint8_t* raw,
```

Comment:
Unsafe ctor moved here and needed friends added.

Reply:
"explicit" is not needed

Reply:
Interestingly, in C++11, explicit does have semantics on multi-argument ctors: it inhibits copy initialization from an initializer list:

https://stackoverflow.com/questions/39122101/explicit-constructor-taking-multiple-arguments/39122237

I'll need to think about what that means for the project internal implementation, but as the function is now private it can't negatively affect the user facing API. Good looking out though. I'll leave it for now until I make a decision about how I feel about the C++11 semantics.

Reply:
Thank you for the explanation.